### PR TITLE
Feat/yarn project type

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### Any background context you want to provide?
We can support Yarn projects in the main app, allow any `yarn` CLI projects to be created as the `yarn` type from now on instead of `npm`.

Previously added projects are unaffected until they are deleted in the UI.
Any new project with a yarn.lock will be created as `yarn` type.

#### Screenshots

Screen Shot 2019-03-16 at 15 54 38

### CODE REVIEW

**Comment by orsagie** [comment]
Do we need the node_modules inside the yarn-package-with-subfolders?

File: help/file.txt [hunk 1]
```diff
@@ -14,13 +14,14 @@ looking for files in following order:
...
-  requirements.txt
   build.gradle
   build.gradle.kts
   build.sbt
+  Pipfile
+  requirements.txt
   Gopkg.lock
   vendor/vendor.json
-  obj/project.assets.json
+  project.assets.json
```
Comment (orsagie) [inline help/file.txt]: why is this changing?

Comment (lili2311) [reply help/file.txt]: We look in both locations https://github.com/snyk/snyk/blob/f5511061847d3ec21c2e49b7afc8bea15393aff6/src/lib/detect.ts#L23-L24

Might as well only show the name of the file like for all others was my thinking.

**Comment by lili2311** [comment]
> yarn-package-with-subfolders

Indeed, yarn lockfiles parsing is not supported on node 4 so we have to fall back to traversing node_modules

**Pull request merged**